Makefile

```diff
@@ -129,6 +129,13 @@ test-%:
 	go test -v -timeout 15m -count 1 --tags test -p 1 ./test/$*
 	$(MAKE) cleanup-e2e
 	
+delete-and-install:
```

Comment:
I couldn't find a make command which will delete and reinstall all control plane components, this was helpful in developing and testing controllers quickly.

Let me know if a similar command exists, i'll remove this 👍

Reply:
Maybe move them into `make image`, and `make e2eapi-image`, under

```
ifdef IMAGE_IMPORT_CMD
        $(IMAGE_IMPORT_CMD) $(IMAGE_NAMESPACE)/$(BINARY_NAME):$(VERSION)
        # here.
endif
```

Reply:
I think `image` and `e2eapi-image` should only be dedicated for creating new image.

For deleting, we should have a separate command like  `delete-numaflow-controller-ux-webhook`

```
delete-numaflow-controller-ux-webhook:
	kubectl -n numaflow-system delete po -lapp.kubernetes.io/component=controller-manager,app.kubernetes.io/part-of=numaflow --ignore-not-found=true
	kubectl -n numaflow-system delete po -lapp.kubernetes.io/component=numaflow-ux,app.kubernetes.io/part-of=numaflow --ignore-not-found=true
	kubectl -n numaflow-system delete po -lapp.kubernetes.io/component=numaflow-webhook,app.kubernetes.io/part-of=numaflow --ignore-not-found=true
	kubectl -n numaflow-system delete po e2e-api-pod --ignore-not-found=true
```

and for creating new image + deleting old image pods
```
reinstall-numaflow-controller-ux-webhook:
	$(MAKE) cleanup-e2e
	$(MAKE) delete-numaflow-controller-ux-webhook
```

Reply:
Let me know how it sounds

Reply:
`image` and `e2eapi-image` are also import images if a local cluster is used. If it's local cluster, there's no reason that you need to restart the pods.

If you look at your changes, all the pod deletion behaviors follow an `image` operation.

Reply:
agreed that `make image` will import the image in the cluster, but we still require pods to be killed, so that we can run the new image.

My goal was to: 
- create a new image
- import into the cluster
- create pods using this new image